mllib/src/main/scala/org/apache/spark/ml/r/GeneralizedLinearRegressionWrapper.scala

```diff
@@ -52,7 +59,15 @@ private[r] class GeneralizedLinearRegressionWrapper private (
   def residuals(residualsType: String): DataFrame = glm.summary.residuals(residualsType)
 
   def transform(dataset: Dataset[_]): DataFrame = {
-    pipeline.transform(dataset).drop(glm.getFeaturesCol)
+    if (rFamily == "binomial") {
+      pipeline.transform(dataset)
+        .drop(PREDICTED_LABEL_PROB_COL)
```

Comment:
This is going to make R models incompatible with other languages when you persist them.  (They already are because of having hidden Pipelines, but that is fixable.)  This, however, will encode special behavior which is only respected when the model is loaded from R, not from other languages.

One option is to encode this in a SQLTransformer.

I'm also worried that these hard-coded columns names will lead to future bug reports about conflicting input column names.

It looks like this same issue appears in other PRs for R, such as [https://issues.apache.org/jira/browse/SPARK-18401].  Given the pervasiveness and that we're in QA right now, I'd recommend we not worry about it for 2.1 and delay fixing it until 2.2.